src/Plutarch/Test/QuickCheck/Modifiers.hs

```diff
@@ -292,6 +307,160 @@ instance Function (GenValue adaMod otherMod) where
   {-# INLINEABLE function #-}
   function = functionMap unwrap rewrap
 
+{- | Represents a change in 'POSIXTime'. The @mod@ argument gives the kind of
+ change, represented by a QuickCheck modifier, while the @n@ argument is a
+ closed (inclusive) upper bound on the magnitude of the change.
+
+ For example, @'TimeDelta' 'Positive' 100@ represents a change in 'POSIXTime'
+ from @1@ to @100@ units, while @'TimeDelta' 'NonPositive' 250@ represents a
+ change in 'POSIXTime' from @0@ to @-250@. Modifiers intended to work with
+ this type are:
+
+ - 'Positive'
+ - 'Negative'
+ - 'NonPositive'
+ - 'NonNegative'
+
+ The instances for 'TimeDelta' reflect this decision: while other modifiers
+ could potentially be useful, this ensures that we have safe behaviour and
+ efficient instances.
+
+ Shrinking a 'TimeDelta' will shrink towards zero: more specifically, it will
+ reduce the magnitude of the change.
+
+ To control what 'TimeDelta' you get, the easiest method is a type signature:
+
+ > forAll arbitrary $ \(delta :: TimeDelta NonNegative 100) -> ...
+
+ @since 2.1.4
+-}
+newtype TimeDelta (mod :: Type -> Type) (n :: Nat)
+  = TimeDelta (mod POSIXTime)
+
+-- | @since 2.1.4
+deriving via
```

Comment:
That would turn `TimeDelta` into a GADT. For what we're aiming for, this is severely overkill, as it would require carrying around two runtime dictionaries.

Reply:
Oh, I didn't know how GADTs work under the hood. But, I don't think this requires GADT,

This will work:
```hs
data TimeDelta (mod :: Type -> Type) (n :: Nat)
  = (Show (mod POSIXTime), Eq (mod POSIXTime)) => TimeDelta (mod POSIXTime)
```